ci/check_runtime_changes.sh

```diff
@@ -30,7 +35,13 @@ FATAL="${red}${block}FATAL${nc}"
 
 
 # show the diff of origin/master and this PR sha
-CHANGED_FILES=$(git diff --name-only ${BASE_COMMIT}...${PR_COMMIT})
+CHANGED_FILES=$(git diff --name-only ${BASE_COMMIT} ${PR_COMMIT} 2>&1 )
+GIT_STATUS=$?
+if (( $GIT_STATUS != 0 ))
+then
+	echo -e "${red}${bold}GIT ERROR${nc}: $CHANGED_FILES"
+	exit 1
+fi
```

Comment:
Can you point me to the explanation of what the status codes of git diff mean? I played around with them locally, and the status code always seems to be `0`

Reply:
The `git diff` failed awhile back when I tested on my fork.  
This only happens when you set the default branch repo in github to some other branch other than `master`

Full details:
When I created a pull-request (with changes to `runtime/`, but no changes to `spec_version` and `impl_version`), I expected the whole script to fail, however it passed for the wrong reason.

`git diff` failed` because I set the default branch of my fork repo to some branch other than `master`, to which `travis-ci` pulls by default. So, when `git diff origin/master..{PR_COMMIT}`, it couldn't find `origin/master` in the git tree, therefore the command fails with exit 1 (which we didn't check). In the script, we only capture the output string (which returns the error messages instead of files that changed in `runtime/` directory.
We match which files are changed in the `runtime/` (empty at this point) and check the spec version (no version changed) so the whole script passed.

Reply:
Makes sense, thanks!